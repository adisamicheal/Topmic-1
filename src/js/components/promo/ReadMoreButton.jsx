// ########## Import Dependencies Here ##########
import React from 'react';
import { string } from 'prop-types';

// ########## Import Screens Here ##########

// ########## Import Components Here ##########

const ReadMoreButton = (props) => {
  const { align } = props;
  return (
    <div className={`read-more-button-${align}`}>
      <p className="read-more-button-text">
        Read More
      </p>
    </div>
  );
}

ReadMoreButton.propTypes = {
  align: string.isRequired,
}

export default ReadMoreButton;