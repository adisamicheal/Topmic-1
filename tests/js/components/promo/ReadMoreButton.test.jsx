import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ReadMoreButton from '../../../../src/js/components/promo/ReadMoreButton';

Enzyme.configure({ adapter: new Adapter() });

describe('testing ReadMoreButton component', () => {
  const wrapper = shallow(<ReadMoreButton align={'center'} />);
  it('renders the ReadMoreButton component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});