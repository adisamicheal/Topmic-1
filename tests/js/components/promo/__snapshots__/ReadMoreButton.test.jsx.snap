// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ReadMoreButton component renders the ReadMoreButton component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReadMoreButton
    align="center"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="read-more-button-text"
      >
        Read More
      </p>,
      "className": "read-more-button-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Read More",
        "className": "read-more-button-text",
      },
      "ref": null,
      "rendered": "Read More",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="read-more-button-text"
        >
          Read More
        </p>,
        "className": "read-more-button-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Read More",
          "className": "read-more-button-text",
        },
        "ref": null,
        "rendered": "Read More",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
